src/rails-route-builder.js

```diff
@@ -8,7 +8,27 @@ class RailsRouteBuilder {
   * @constructor
   */
   constructor () {
-    this.pathBuilder = new PathBuilder()
+    this.pathBuilder  = new PathBuilder()
+    this.chainedPaths = []
+  }
+
+  /**
+  * Defines a namespace to be used in the next route of the chain
+  * @param {string} resource - the name of the resource to be used as namespace
+  * @param {integer} id - the ID of the resource, can be left empty
+  * @example
+  * const routes = new RailsRouteBuilder
+  * routes.resource('users', 1).list('blogPosts')
+  * //=> { path: '/users/1/blog_posts', params: {} }
+  */
+  resource (resource, id = null) {
+    const newInstance = Object.assign({}, this)
+    const path        = id ? `${resource}/${id}` : resource
+
+    newInstance.chainedPaths.push(path)
+    Object.setPrototypeOf(newInstance, RailsRouteBuilder.prototype)
```

Comment:
Reading about `Object.setPrototypeOf` I found this warning:

> Warning: Changing the [[Prototype]] of an object is, by the nature of how modern JavaScript engines optimize property accesses, a very slow operation, in every browser and JavaScript engine. The effects on performance of altering inheritance are subtle and far-flung, and are not limited to simply the time spent in Object.setPrototypeOf(...) statement, but may extend to any code that has access to any object whose [[Prototype]] has been altered. If you care about performance you should avoid setting the [[Prototype]] of an object. Instead, create a new object with the desired [[Prototype]] using Object.create().

more info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf

I don't know if it's a problem for this project. Did you consider it?

Reply:
Hmmmm, nice catch!
I'm gonna try to find a better approach for this, thanks 😄